runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8182,7 +8182,8 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
             TR_ASSERT(!that->_methodBeingCompiled->isOutOfProcessCompReq(), "JITServer should not change options passed by client");
 #endif /* defined(J9VM_OPT_JITSERVER) */
             bool aotCompilationReUpgradedToWarm = false;
-            if (that->_methodBeingCompiled->_useAotCompilation)
+            if (that->_methodBeingCompiled->_useAotCompilation
+                && !TR::Options::getCmdLineOptions()->getOption(TR_FullSpeedDebug))
```

Comment:
I think a comment is warranted here explaining why we exclude FSD from doing warm AOT

Reply:
Added in https://github.com/eclipse/openj9/pull/11554/commits/b8505f40ed73b49c9551f75b3ef4385e2761a029